src/monitor/node_active_protocol.c

```diff
@@ -2448,7 +2441,8 @@ synchronous_standby_names(PG_FUNCTION_ARGS)
 		int count = list_length(syncStandbyNodesGroupList);
 
 		if (count == 0 ||
-			IsCurrentState(primaryNode, REPLICATION_STATE_WAIT_PRIMARY))
+			(primaryNode != NULL && IsCurrentState(primaryNode,
+												   REPLICATION_STATE_WAIT_PRIMARY)))
```

Comment:
Given the following current definition of `IsCurrentState`, it looks to me that this change is not needed either:

``` c
bool
IsCurrentState(AutoFailoverNode *pgAutoFailoverNode, ReplicationState state)
{
	return pgAutoFailoverNode != NULL &&
		   pgAutoFailoverNode->goalState == pgAutoFailoverNode->reportedState &&
		   pgAutoFailoverNode->goalState == state;
}
```